### PR TITLE
Add myplaces list in profile

### PR DESCRIPTION
**Pull request by KostyaKobel**
# Checklist:

- [ ] Added specs for all new ruby code
- [x] Green rspec run
- [x] Green rubocop
- [ ] PR review from teammate

# What?
Add my places list in a Profile Page for Owner Place. 

## Why?
These changes will add each Owner who has Place  opportunity to see a list of their places.

## How?
When creating a new place user will be Owner this place and for him will be add opportunity see him places list on My Places page.

## Testing?
Yes, I did.

## Screenshots (optional)
place

### CODE REVIEW

File: app/frontend/stylesheets/header.scss [hunk 1]
```diff
@@ -39,4 +39,4 @@ header {
       }
     }
   }
-}
+}
```
Comment (andrjuha01) [inline app/frontend/stylesheets/header.scss:42]: new line

File: app/models/place.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Place < ApplicationRecord
+  belongs_to :user, optional: true
```
Comment (andrjuha01) [inline app/models/place.rb:2]: so place can be created without user?
I don't think that it is right

File: app/models/user.rb [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable, :confirmable, :omniauthable,
          omniauth_providers: [:google_oauth2]
   enum role: { consumer: 'consumer', owner: 'owner' }
-  validates :first_name, presence: true, length: { minimum: 2, maximum: 50 }
+
+  validates :first_name, length: { minimum: 2, maximum: 50 }
   validates :role, presence: true
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  VALID_EMAIL_REGEX = /\A[^@\s]+@[^@\s]+\z/
```
Comment (andrjuha01) [inline app/models/user.rb:12]: Use devise validations for email and password

File: app/views/places/my_places.html.slim [hunk 4]
```diff
@@ -0,0 +1,33 @@
+- provide :title, "My Places"
+
+section.my-places
+	= render 'shared/header'
+	.container
```
Comment (andrjuha01) [inline app/views/places/my_places.html.slim:5]: why tabs are too wide now?

File: app/controllers/places_controller.rb [hunk 5]
```diff
@@ -1,11 +1,30 @@
...
+  before_action :place_find, only: [:show]
+
   def index
     @places = Place.where(status: :created).paginate(page: params[:page], per_page: PLACES_PER_PAGE)
   end
 
-  def show
+  def show; end
+
+  def new; end
```
Comment (andrjuha01) [inline app/controllers/places_controller.rb:12]: do you use new and edit endpoints?

Comment (KostyaKobel) [reply app/controllers/places_controller.rb:12]: Yes I do.I add rspec tests where I need add these actions in controller.

**Review by Hunter2065 — COMMENTED** [review]
image
you must change link to editing place

**Review by Hunter2065 — COMMENTED** [review]
image
will be good if you can add greater retreat to bottom

Fixed

**Review by Hunter2065 — COMMENTED** [review]
image
it's not good

**Review by Hunter2065 — CHANGES_REQUESTED** [review]
image
change position filters' dropdown-list to center

image
it's not good(
image
it's too, please make it more in responsible design

image
my phone will not approve that(

Fixed

File: Gemfile [hunk 6]
```diff
@@ -19,6 +19,8 @@ gem 'turbolinks', '~> 5'
 gem 'jbuilder', '~> 2.7'
 # Slim HTML
 gem 'slim', '~> 4.1'
+gem 'carrierwave', '~> 2.2', '>= 2.2.2'
+gem 'mini_magick', '~> 4.11'
```
Comment (Hunter2065) [inline Gemfile:23]: do you use  these?

File: spec/views/places/_my_place.html.slim_spec.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  context 'when template rendered' do
+    it 'infers the controller path' do
+      expect(controller.controller_path).to eq('places')
+    end
+
+    it 'infers the controller action' do
+      expect(controller.request.path_parameters[:action]).to eq('my_place')
+    end
+  end
+end
```
Comment (Hunter2065) [inline spec/views/places/_my_place.html.slim_spec.rb:13]: I think this test for template, not for partial, 'cause it's part of template

**Review by Hunter2065 — CHANGES_REQUESTED** [review]

**Review by Hunter2065 — COMMENTED** [review]
image
your will_paginate must be like this

File: app/frontend/stylesheets/header.scss (hunk 1 above)
Comment (KostyaKobel) [reply app/frontend/stylesheets/header.scss]: Fixed

File: app/models/place.rb (hunk 2 above)
Comment (KostyaKobel) [reply app/models/place.rb]: Fixed

File: app/models/user.rb (hunk 3 above)
Comment (KostyaKobel) [reply app/models/user.rb]: Fixed

File: Gemfile (hunk 6 above)
Comment (KostyaKobel) [reply Gemfile]: Fixed

Comment (KostyaKobel) [reply Gemfile]: Fixed

File: spec/views/places/_my_place.html.slim_spec.rb (hunk 7 above)
Comment (KostyaKobel) [reply spec/views/places/_my_place.html.slim_spec.rb]: Fixed

**Review by Hunter2065 — CHANGES_REQUESTED** [review]
image
Opps, when user hasn't places page must looks like this:
image

File: app/frontend/stylesheets/places.scss [hunk 8]
```diff
@@ -300,37 +300,298 @@ $btn-color: #91CB82;
...
+    }
+  }
+
+  a {
+      color: var(--bs-light);
+      text-decoration: none;
+    }
+
+  .hovc:hover {
+    color: #505050 !important;
```
Comment (Hunter2065) [inline app/frontend/stylesheets/places.scss:368]: i think using important is not a good practice

File: app/frontend/stylesheets/places.scss [hunk 9]
```diff
@@ -300,37 +300,298 @@ $btn-color: #91CB82;
...
+            width: 106%;
+          }
+          @media (min-width: 990px) and (max-width: 1200px ) {
+            font-size: 1em;
+            width: 106%;
+          }
+          @media (min-width: 1200px) and (max-width: 1400px ) {
+            width: 104%;;
+          }
+          @media (min-width: 1400px) and (max-width: 1920px ) {
```
Comment (Hunter2065) [inline app/frontend/stylesheets/places.scss:522]: it's tooo many media-requests), can we use less requests?) just we use bootstrap for responsible-adaptive design

File: config/initializers/carrier_wave.rb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+if Rails.env.production?
+  CarrierWave.configure do |config|
+    config.fog_credentials = {
+      provider: 'AWS',
+      aws_access_key_id: ENV['S3_ACCESS_KEY'],
+      aws_secret_access_key: ENV['S3_SECRET_KEY'],
+    }
+    config.fog_directory = ENV['S3_BUCKET']
+  end
+end
```
Comment (Hunter2065) [inline config/initializers/carrier_wave.rb:10]: remove this file!!

**Review by Hunter2065 — CHANGES_REQUESTED** [review]
resolve some issues in comments

File: app/frontend/stylesheets/places.scss (hunk 9 above)
Comment (KostyaKobel) [reply app/frontend/stylesheets/places.scss:522]: If I need use it I use.

File: config/initializers/carrier_wave.rb (hunk 10 above)
Comment (KostyaKobel) [reply config/initializers/carrier_wave.rb]: Fixed

File: app/frontend/stylesheets/places.scss (hunk 8 above)
Comment (KostyaKobel) [reply app/frontend/stylesheets/places.scss:368]: It good practice when I need use it.

Comment (andrjuha01) [reply app/frontend/stylesheets/places.scss:368]: you can specify better selector to not use !important. Please try to fix it

File: app/frontend/stylesheets/places.scss (hunk 9 above)
Comment (Hunter2065) [reply app/frontend/stylesheets/places.scss:522]: why only you have so?

File: app/models/user.rb [hunk 11]
```diff
@@ -1,17 +1,18 @@
...
+
+  validates :first_name, length: { minimum: 2, maximum: 50 }
   validates :role, presence: true
-  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
   validates :email, presence: true,
-                    format: { with: VALID_EMAIL_REGEX },
+                    format: { with: email_regexp },
                     uniqueness: { case_sensitive: false }
-  validates :password, presence: true, length: { minimum: 6, maximum: 256 }
+  validates :password, length: { minimum: 6, maximum: 256 }
```
Comment (Hunter2065) [inline app/models/user.rb:15]: presence must be, 'cause without it the errors do not return in flash-messages

**Review by Hunter2065 — COMMENTED** [review]
some comment

File: app/models/user.rb (hunk 11 above)
Comment (KostyaKobel) [reply app/models/user.rb:15]: If we use minimum and maximum length we don't need use validation  check on presence.And flash-messages present for registration User.

File: app/frontend/stylesheets/places.scss (hunk 8 above)
Comment (KostyaKobel) [reply app/frontend/stylesheets/places.scss]: Fixed

**Review by Hunter2065 — COMMENTED** [review]
image
change please color in comment

**Review by Hunter2065 — CHANGES_REQUESTED** [review]
image
?

**Review by andrjuha01 — APPROVED** [review]

**Review by Hunter2065 — APPROVED** [review]
I think all is good now

**Pull request merged**